### PR TITLE
Drop k8s 1.28 and 1.29 metal variants

### PR DESCRIPTION
**Pull request by yeazelm**
**Issue number:**

Related to https://github.com/bottlerocket-os/bottlerocket/issues/3794 

**Description of changes:**

This removes the metal 1.28 and 1.29 variants. Kubernetes 1.28 went end-of-life on Oct 28 and EKS end-of-life for 1.28 is November 26th and these variants are no longer supported. For more information on end of life of metal variants, please see https://github.com/bottlerocket-os/bottlerocket/issues/3794

Testing done:

Default build cargo-make succeeds and `cargo make -e BUILDSYS_VARIANT=meta-dev` still works and builds the dev variant of metal.

**Testing done:**

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,8 +26,6 @@ members = [
     "variants/aws-k8s-1.30-nvidia",
     "variants/aws-k8s-1.31-nvidia",
     "variants/metal-dev",
-    "variants/metal-k8s-1.28",
```
Comment (ginglis13) [inline Cargo.toml:34]: I think we should drop the corresponding settings-plugins for these variants (looks like I'd missed that in the 1.27 drop amidst the core kit shift 😅): https://github.com/bottlerocket-os/bottlerocket/blob/b681183a68201d50ba0bd408637789a597b86fa2/packages/settings-plugins/settings-plugins.spec#L98-L109

File: variants/README.md [hunk 2]
```diff
@@ -446,6 +433,20 @@ It supported self-hosted clusters.
...
+
+This variant is compatible with Kubernetes 1.28, 1.29, 1.30, and 1.31 clusters.
+
+### metal-k8s-1.29: Metal Kubernetes 1.29 node
+
+The [metal-k8s-1.29](metal-k8s-1.29/Cargo.toml) variant includes the packages needed to run a Kubernetes node on bare metal.
+It supports self-hosted clusters.
+
+This variant is compatible with Kubernetes 1.29, 1.30, 1.31, and 1.32 clusters.
+
```
Comment (ginglis13) [inline variants/README.md]: Should use past tense for these now that they are deprecated

Comment (yeazelm) [reply variants/README.md]: Nice catch!

File: Cargo.toml (hunk 1 above)
Comment (yeazelm) [reply Cargo.toml:34]: That makes sense, I removed the defaults and plugins for metal-k8s.

**Review by ginglis13 — APPROVED** [review]

**Review by ginglis13 — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]

**Pull request merged**